src/mod_muc_room.erl

```diff
@@ -4027,12 +4020,11 @@ element_size(El) ->
 
 -spec store_room(state()) -> ok.
 store_room(StateData) ->
-    if (StateData#state.config)#config.persistent ->
-	    mod_muc:store_room(StateData#state.server_host,
+    case mod_muc:store_room(StateData#state.server_host,
 			       StateData#state.host, StateData#state.room,
-			       make_opts(StateData));
-      true ->
-        ok
+			       make_opts(StateData)) of
+       {true} ->
+          ok
```

Comment:
what's going on here?

Reply:
Switched this from the if statement that only allows the storage of the room when persistent to allowing storage regardless of state and then pattern matching to a case statement to return what it was expecting before.  Defaults to `error`, I believe

Reply:
nit but seems like you're just using the case as an if statement? Seems unnecessarily clunky, why not just change the condition of the if statement?

Reply:
Different paradigms for erlang.  http://erlang.org/pipermail/erlang-questions/2008-March/033673.html